Source/Peripheral.swift

```diff
@@ -326,7 +326,8 @@ public class Peripheral {
      */
     public func readValue(for characteristic: Characteristic) -> Observable<Characteristic> {
         return .create { observer in
-            let disposable = self.monitorValueUpdate(for: characteristic).take(1).subscribe(observer)
+            let disposable = self.ensureValidPeripheralState(for: self.monitorValueUpdate(for: characteristic).take(1))
+                .subscribe(observer)
             self.peripheral.readValue(for: characteristic.characteristic)
```

Comment:
I think we should make sure that readValue will not be called as a side effect in situation where subscribe call above detects any invalid state which prohibits BLE operations. `readValue` function should be called after `monitorValueUpdate` subscription.